pkg/registry/core/pod/strategy.go

```diff
@@ -425,6 +446,52 @@ func ExecLocation(
 	return streamLocation(getter, connInfo, ctx, name, opts, opts.Container, "exec")
 }
 
+// DebugLocation returns the debug URL for a pod container.
+func DebugLocation(
+	getter ResourceGetter,
+	connInfo client.ConnectionInfoGetter,
+	ctx genericapirequest.Context,
+	name string,
+	opts *api.PodDebugOptions,
+) (*url.URL, http.RoundTripper, error) {
+	if !utilfeature.DefaultFeatureGate.Enabled(features.DebugContainers) {
+		return nil, nil, errors.NewBadRequest("debug containers feature disabled")
+	}
+
+	pod, err := getPod(getter, ctx, name)
+	if err != nil {
+		return nil, nil, err
+	}
+
+	if opts.Container == "" {
+		return nil, nil, errors.NewBadRequest(fmt.Sprintf("a new container name must be specified for pod %s", name))
+	}
+	if podHasContainerWithName(pod, opts.Container) {
+		return nil, nil, errors.NewBadRequest(fmt.Sprintf("container %s already exists for pod %s", opts.Container, name))
+	}
+
+	nodeName := types.NodeName(pod.Spec.NodeName)
+	if len(nodeName) == 0 {
+		// If pod has not been assigned a host, return an empty location
+		return nil, nil, errors.NewBadRequest(fmt.Sprintf("pod %s does not have a host assigned", name))
+	}
+	nodeInfo, err := connInfo.GetConnectionInfo(nodeName)
+	if err != nil {
+		return nil, nil, err
+	}
+	params := url.Values{}
+	if err := streamParams(params, opts); err != nil {
+		return nil, nil, err
+	}
+	loc := &url.URL{
+		Scheme:   nodeInfo.Scheme,
+		Host:     net.JoinHostPort(nodeInfo.Hostname, nodeInfo.Port),
+		Path:     fmt.Sprintf("/podDebug/%s/%s/%s", pod.Namespace, pod.Name, opts.Container),
```

Comment:
most of the other two word kubelet paths use mixed case (e.g. portForward, containerLogs), but then there's "runningpods" so perhaps that's an old convention.